### PR TITLE
Add support for JUnit 4

### PR DESCRIPTION
**Pull request by Croway**

### CODE REVIEW

File: pom.xml
Comment (gunnarmorling) [reply pom.xml:90]: > spring-test pulls both libraries

Interesting one; note they mark these dependencies as optional.

>  do you have any project for integration testing? 

Yes, you could have a look at the examples repo here: https://github.com/moditect/jfrunit-examples

>  I can manually test this version with both libraries, and check if everything runs smoothly

That's a great idea. Thanks a lot!

**Review by gunnarmorling — APPROVED** [review]
LGTM overall. I think we can merge this, pending your feedback from running this with the examples repo.

**Comment by gunnarmorling** [comment]
Ah, one thing only: can you rework the commit using an email linked to your GH account? Thx!

**Comment by Croway** [comment]
> LGTM overall. I think we can merge this, pending your feedback from running this with the examples repo.

Thanks!, examples runs as expected (there is 1 failure but I think that you want to have that, using EAGER instad of LAZY I managed to execute successfully all tests) I tested both junit4/5 dependencies marked as optional and not marked as optional, the tests are always successful and the build complete as expected. Do you want to put junit dependency as optional? or I can leave as it is and just edit the email?

**Review by gunnarmorling — APPROVED** [review]

**Comment by gunnarmorling** [comment]
Applied. Thank you so much, @Croway!

**Comment by gunnarmorling** [comment]
Also sent you an invite to join as a committer, if you like. If you're up for it, the only rule really is that we don't merge our own changes, i.e. there should always be four eyes on each change, with a reviewer merging PRs. The exception are trivial typo fixes and such, just commit these things straight to `main`.

**Comment by Croway** [comment]
Thanks, I'm honored!

**Pull request merged**
